packages/utils/src/translation-helpers.ts

```diff
@@ -68,7 +74,14 @@ export function applyTranslation(
   });
 
   traverse(blocks).forEach(function (el) {
-    // TODO: localized custom components inputs
+    const path = this.path?.join('#');
+    if (translation[`blocks.${path}`]) {
+      this.update({
+        ...el,
+        [locale]: translation[`blocks.${path}`].value,
+      });
```

Comment:
This feels incorrect,  we should be only updating the bindings to point to the state.translations property, same as the next few lines